src/testing.py

```diff
@@ -157,7 +157,7 @@ def __setitem__(self, key, value):
     k = self.canonicalize(key)
     # Support single-insertion only, the intended usage would be a bug if there
     # were multiple insertions of the same key.
-    assert k not in self.arr, 'Double insertion of key %s' % k
```

Comment:
I thought `%s` would `str`-convert args anyway, what's the difference here?

Reply:
This is needed if you want to format a tuple (which this is), otherwise formatting will raise and exception that there are not enough % chars in the string.